pallets/ajuna-awesome-avatars/src/lib.rs

```diff
@@ -299,6 +300,10 @@ pub mod pallet {
 		pub fn mint(origin: OriginFor<T>) -> DispatchResult {
 			let player = ensure_signed(origin)?;
 			ensure!(Self::mint_available(), Error::<T>::MintUnavailable);
+			ensure!(
+				Self::players(&player).len() < MAX_AVATARS_PER_PLAYER as usize,
+				Error::<T>::MaxOwnershipReached
+			);
```

Comment:
Early return here is important because `Self::random_dna()` mutates `Nonce` storage -> this means `assert_noop!` always fails as it detects changes in storage before and after.